app/src/main/java/com/github/se/signify/MainActivity.kt

```diff
@@ -42,14 +47,41 @@ import com.github.se.signify.ui.theme.SignifyTheme
 import kotlinx.coroutines.flow.MutableStateFlow
 
 class MainActivity : ComponentActivity() {
+
+  private lateinit var sharedPreferences: SharedPreferences
+
   override fun onCreate(savedInstanceState: Bundle?) {
     super.onCreate(savedInstanceState)
+
+    // Initialize SharedPreferences
+    @Suppress("SonarCloud:UNENCRYPTED_SHARED_PREF")
+    sharedPreferences = getSharedPreferences("theme_prefs", Context.MODE_PRIVATE)
+
+    // Get the saved theme state (default is false for light mode)
+    val savedTheme = sharedPreferences.getBoolean("is_dark_theme", false)
+
     setContent {
-      SignifyTheme {
+      var isDarkTheme by remember { mutableStateOf(savedTheme) }
+
+      // Save theme state when toggled
+      fun saveThemePreference(isDark: Boolean) {
```

Comment:
This function should be in Utils.kt or at least at the bottom of the class.

Reply:
Thanks for the suggestion! However, in our project, utils are specifically reserved for UI-related components. Instead of moving the preference logic to utils, I decided to remove the saveThemePreference function and directly handle the preference logic in MainActivity. Additionally, I moved the preference strings to strings.xml to centralize them, align with Android best practices, and allow for easier localization or modification."